internal/uvm/network.go

```diff
@@ -59,6 +61,16 @@ func (uvm *UtilityVM) SetupNetworkNamespace(ctx context.Context, nsid string) er
 		return err
 	}
 
+	// Sort the endpoints.
+	// In some scenarios, multiple endpoints may be added to a pod. When this happens
+	// we need to know which endpoint to add first as the eth0 interface in the guest.
+	// Since we don't have CNI results here, instead look if any endpoints have the
+	// name eth0 at the end. If so, add that endpoint first. Otherwise this is a noop.
+	less := func(i, j int) bool {
+		return strings.HasSuffix(endpoints[i].Name, "eth0")
+	}
+	sort.Slice(endpoints, less)
```

Comment:
Since this isn't really a sort, does it make more sense to use [`slices.IndexFunc`](https://pkg.go.dev/slices#IndexFunc) to find the location of the eth0 interface first then move it to the beginning?
that way we can also warn/error if none of them end in eth0?

Reply:
It's not an error or even really a warning if there are none that end in -eth0. Our nat cni for example does not use -ethX in the naming scheme. I'm fine using `slices.IndexFunc` if we want, but for now the only use case of this is an array with two endpoints so I don't think the sort should be problematic.

Reply:
if its not noteworthy, then its fine as is (though `less` here doesn't implement a true partial ordering cause reflexivity/symmetry, though that shouldn't matter unless the CNI messes up somehow)
(also, i think [`slices.SortFunc`](https://pkg.go.dev/slices#SortFunc) is recommended in the docs for `sort.Slice`?)

Reply:
Updated to `slices.SortFunc` and also added some unit tests